### ISSUE TITLE
[BUG] Cannot load keras autlog model from registry due to model/model/ directory doubling

### ISSUE THREAD

**Issue by Dr-DaDe** [mlflow/mlflow#2641]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS Catalina
- **MLflow installed from (source or binary)**: both via pip 1.7.0 and pip+git 1.7.3dev0
- **MLflow version (run ``mlflow --version``)**: 1.7.0 and 1.7.3.dev0
- **Python version**: 3.7.6

I installed a working model registry on a remote server using postgresql as a basis. As for now, I have not encountered any problems with logging models to the registry both via the UI interface and via the API:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/60467599/77654810-d2645600-6f71-11ea-8a71-9d8e8db59e95.png">

I have no problems in loading models via their run_id using e.g
```
from mlflow.keras import load_model
model = load_model("runs:/e510981424cc4d7e9b91d7ceb15c8098/model/model")
```
However, if I try to load a model from the registry, I encounter a bug

```
load_model("models:/MyKerasModel/3")

Traceback (most recent call last):
  File "mini.py", line 6, in <module>
    load_model("models:/MyKerasModel/3")
  File "/Users/daniel.dercks/opt/anaconda3/envs/berner_macos_nogpu/lib/python3.7/site-packages/mlflow/keras.py", line 391, in load_model
    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
  File "/Users/daniel.dercks/opt/anaconda3/envs/berner_macos_nogpu/lib/python3.7/site-packages/mlflow/utils/model_utils.py", line 26, in _get_flavor_configuration
    RESOURCE_DOES_NOT_EXIST)
mlflow.exceptions.MlflowException: Could not find an "MLmodel" configuration file at "/var/folders/jx/02yypg8x5wjdhhr54xcmzhy87m0bvt/T/tmpmo0g0x22/"
```

Interestingly, if I look inside the folder of interest, I do see that it countains a `model` subdirectory  which contains the model of interest.
```
~$ ls /var/folders/jx/02yypg8x5wjdhhr54xcmzhy87m0bvt/T/tmpmo0g0x22/
model
~$ ls /var/folders/jx/02yypg8x5wjdhhr54xcmzhy87m0bvt/T/tmpmo0g0x22/model/
MLmodel    conda.yaml data
```

It seems as if mlflow forgets to look inside the /model subfolder. Unfortunately, I cannot tell it to do so because if I run

```
load_model("models:/MyKerasModel/3/model")
```
I get
```
mlflow.exceptions.MlflowException: Not a proper models:/ URI: models:/MyKerasModel/3/model. Models URIs must be of the form 'models:/<model_name>/<version or stage>'.
```

If it is of any interest: models were stored 'automatically' via the `mlflow.keras.autolog()` functionality and I used the "Register Model" button on the UI to publish the model inside the model registry.  
Interestingly, there seems to be  a potentially peculiar `model/model/...` directory structure. So maybe it's keras.autolog() who is the culprit and should not produce a model folder within a model folder
<img width="1303" alt="image" src="https://user-images.githubusercontent.com/60467599/77656867-c8902200-6f74-11ea-9916-737d51eafe26.png">

Note also how my `load_model("runs:/...`command from above also explicitly has to look inside models/models

**EDIT** In investigated the double folder structre more thoroughly.
So in the artifact listing I showed above, I can both click on the main folder "model" and click "Register Model" _and_ I can click on the subfolder "model/model" and click "Register Model". 

If I do the first, I encounter the above error, I I do the second, I don't.

So the solution, to me, seems to be that either or both of the following should be fixed
- Keras.autolog should not produce the model/model double structure
- The "Register Model" should not be clickable unless the folder I selected really is a proper model folder with a proper MLModel definition file inside.

**EDIT2** The bug also appears if I don't use the UI but use the actual `mlflow.keras.log_model` method:

```
In[2]: import mlflow
In[3]: from mlflow.keras import load_model
In[4]: mlflow.set_tracking_uri("http://ki01hb.hmmh.ag:4999")
In[5]: model = load_model("runs:/3289a926f0164ee5bf114405e7fc6723/model/model")
Using TensorFlow backend.
[... some tensorflow warnings ... ]
In[6]: mlflow.keras.log_model(model, "", registered_model_name="MyVeryFirstModel")
Successfully registered model 'MyVeryFirstModel'.
Created version '1' of model 'MyVeryFirstModel'.
In[7]: model2 = load_model("models:/MyVeryFirstModel/1")
Traceback (most recent call last):
  File "/Users/daniel.dercks/opt/anaconda3/envs/berner_macos_nogpu/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3331, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-f1fb653daf70>", line 1, in <module>
    model2 = load_model("models:/MyVeryFirstModel/1")
  File "/Users/daniel.dercks/opt/anaconda3/envs/berner_macos_nogpu/lib/python3.7/site-packages/mlflow/keras.py", line 391, in load_model
    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
  File "/Users/daniel.dercks/opt/anaconda3/envs/berner_macos_nogpu/lib/python3.7/site-packages/mlflow/utils/model_utils.py", line 26, in _get_flavor_configuration
    RESOURCE_DOES_NOT_EXIST)
mlflow.exceptions.MlflowException: Could not find an "MLmodel" configuration file at "/var/folders/jx/02yypg8x5wjdhhr54xcmzhy87m0bvt/T/tmpha64w5wy/"
In[8]: import os
  ...: os.listdir("/var/folders/jx/02yypg8x5wjdhhr54xcmzhy87m0bvt/T/tmpha64w5wy/")
Out[8]: ['model']
```

**Comment by andychow-db** [comment]
**MLflow Version**: mlflow, version 1.7.3.dev0

I tried replicating the issue but it appears to work. The output below is from my terminal. It shows the ability to load a model from both runs and registered models. 

```
>>> model = load_model("runs:/f6c4152f99564883aebd0118674c6961/model")
>>> model
<keras.engine.sequential.Sequential object at 0x7ff430fc1eb8>
>>> registered_model = load_model("models:/keras-example/1") 
>>> registered_model
<keras.engine.sequential.Sequential object at 0x7ff460ad6780>
```

**Comment by Dr-DaDe** [comment]
> **MLflow Version**: mlflow, version 1.7.3.dev0
> **OS**: Mojave 10.14.6
> 
> I tried replicating the issue but it appears to work for me. I used the keras example [link](https://github.com/mlflow/mlflow/blob/master/examples/keras/train.py). The output below is from my terminal. It shows the ability to load a model from both runs and registered models.
> 
> ```
> >>> model = load_model("runs:/f6c4152f99564883aebd0118674c6961/model")
> >>> model
> <keras.engine.sequential.Sequential object at 0x7ff430fc1eb8>
> >>> registered_model = load_model("models:/keras-example/1") 
> >>> registered_model
> <keras.engine.sequential.Sequential object at 0x7ff460ad6780>
> ```
> 
> <img alt="Screen Shot 2020-04-08 at 12 32 22 PM" width="1799" src="https://user-images.githubusercontent.com/58712524/78825557-0ccbeb00-7995-11ea-935d-e442834990b7.png">
> 
> Do you see any differences in the steps taken?

The important questions are: How did you log the model to run f6c4152f99564883aebd0118674c6961 and how did you log it to the model repository?

**Comment by andychow-db** [comment]
@Dr-DaDe,

To create the model, I clicked the "Register Model" button in the UI .

I am using `keras.autolog()` also. https://github.com/mlflow/mlflow/blob/master/examples/keras/train.py#L17

Rereading your description, I think our difference is that I'm not explicitly calling `log_model`. Autolog saves the model for me.

**Comment by Dr-DaDe** [comment]
Ok, I investigated further:
Running https://github.com/mlflow/mlflow/blob/master/examples/keras/train.py indeed works as intended:

![image](https://user-images.githubusercontent.com/60467599/78916711-30269100-7a8e-11ea-933f-5ab4217491c5.png)

**However** by simply changing the tracking_uri to a remote server with one extra line
`mlflow.set_tracking_uri("http://<...>:5000")`
results in a model/model structure

![image](https://user-images.githubusercontent.com/60467599/78916873-66fca700-7a8e-11ea-8596-81874febec9a.png)

As you can see, I am using an ftp artifact root (if that is of any importance here)

**Comment by andychow-db** [comment]
Thank you for digging into it further. I'm going to look at trying to reproduce the issue using an FTP server later today.  

---
I just tried it with an http remove server and it appears to function correctly. 

I spun on a local server using the command `mlflow ui --backend-store-uri sqlite:///mlflow.db --default-artifact-root /Users/Andy.Chow/Workspace/mlflow-issue-2641/artifact-root`

Then I added `mlflow.set_tracking_uri("http://localhost:5000")` to `train.py` and then ran it. It did not change how it was stored.
